geo-types/src/point.rs

```diff
@@ -232,10 +232,10 @@ impl<T: CoordNum> Point<T> {
     /// # Examples
     ///
     /// ```
-    /// use geo_types::{Coordinate, Point};
+    /// use geo_types::{coord, Point};
     ///
-    /// let point = Point(Coordinate { x: 1.5, y: 0.5 });
-    /// let dot = point.dot(Point(Coordinate { x: 2.0, y: 4.5 }));
+    /// let point = Point(coord! { x: 1.5, y: 0.5 });
```

Comment:
I would prefer
```
let point = Point::new(1.5, 0.5);
```
in doc comments.

Reply:
@pka seems like it would create more problems than solve -- https://github.com/georust/geo/runs/5472344274?check_suite_focus=true -- so reverting. Let's keep changes to a minimum per above, and consider other changes separately.

```
error[E0283]: type annotations needed for `geo_types::Point<{float}>`
  --> src/point.rs:237:13
   |
6  | let point = Point::new(1.5, 0.5);
   |     -----   ^^^^^^^^^^ cannot infer type for type `{float}`
   |     |
   |     consider giving `point` the explicit type `geo_types::Point<{float}>`, where the type parameter `{float}` is specified
   |
```

Reply:
How about using the `geo::point! { x: , y: }` macro?

Reply:
But yeah I'm fine keep it as you have written since you just did a find/replace, and this is unrelated

Reply:
@frewsxcv I agree that we should refactor it further in a subsequent PR, flattening macro usage. Feel free to r plus it when ready :)